# Indian Food Truck Market Analysis in Bangalore

## Notebook Goal

This notebook analyzes restaurant data from Zomato to evaluate market conditions and identify opportunities for launching an Indian food truck in Bangalore.

## Main Objectives
- Load and inspect the Zomato dataset
- Identify relevant restaurant and customer information
- Clean and prepare the dataset for analysis
- Analyze patterns related to competition, pricing, and customer ratings
- Identify potential market opportunities

## 1. Project Setup

### Project Context

This project aims to evaluate the feasibility of launching an Indian food truck in Bangalore using a data-driven market research approach.

The analysis is structured into two complementary components:

- **Market & Demand Analysis (API-based)**  
  Focused on identifying high-traffic areas such as metro stations and tech parks using external APIs.

- **Restaurant & Competition Analysis (this notebook)**  
  Focused on analyzing existing restaurant data from Zomato, including competition levels, customer ratings, pricing patterns, and market positioning.

This notebook specifically contributes to understanding the competitive landscape and customer behavior in the Bangalore restaurant market.

---

### Role of This Notebook

This notebook focuses on analyzing the restaurant ecosystem in Bangalore using Zomato data.

The objective is to understand how the market behaves by evaluating:

- competition density (number of restaurants per area)  
- customer satisfaction (ratings and reviews)  
- pricing patterns  
- distribution of restaurant types and cuisines  

This analysis helps identify patterns, saturation levels, and potential gaps where a food truck could be strategically positioned.

---

### Key Analytical Questions

- Which areas in Bangalore have the highest concentration of restaurants?  
- How strong is the competition in different areas?  
- What are the average ratings and customer satisfaction levels?  
- How does pricing vary across locations?  
- Are there areas with high activity but potential gaps in affordable or quick-service options?  

---

### Data Source

The primary dataset used in this notebook is the Zomato dataset, which includes:

- restaurant name and location  
- cuisine types  
- average cost for two people  
- ratings and number of votes  
- restaurant category (e.g., quick bites, casual dining)

This dataset enables both quantitative analysis (counts, averages, distributions) and market pattern identification.

---

### Analytical Objective

The goal of this notebook is to extract insights about:

- competition density across different areas  
- customer perception through ratings  
- pricing distribution and affordability  
- potential market gaps  

These insights will contribute to identifying the most promising areas to launch an Indian food truck, especially when combined with high-traffic location data from external APIs.

## 2. Import Libraries

In [1]:
# Import core libraries for data analysis
import pandas as pd
import numpy as np

# improve display in notebook
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', 100)

## 3. Load Dataset

In [2]:
# Load dataset from current folder
df = pd.read_csv("zomato.csv")

# Preview first rows
df.head()

,url,address,name,online_order,book_table,rate,votes,phone,location,rest_type,dish_liked,cuisines,approx_cost(for two people),reviews_list,menu_item,listed_in(type),listed_in(city)
0,https://www.zomato.com/bangalore/jalsa-banasha...,"942, 21st Main Road, 2nd Stage, Banashankari, ...",Jalsa,Yes,Yes,4.1/5,775,080 42297555\r\n+91 9743772233,Banashankari,Casual Dining,"Pasta, Lunch Buffet, Masala Papad, Paneer Laja...","North Indian, Mughlai, Chinese",800,"[('Rated 4.0', 'RATED\n A beautiful place to ...",[],Buffet,Banashankari
1,https://www.zomato.com/bangalore/spice-elephan...,"2nd Floor, 80 Feet Road, Near Big Bazaar, 6th ...",Spice Elephant,Yes,No,4.1/5,787,080 41714161,Banashankari,Casual Dining,"Momos, Lunch Buffet, Chocolate Nirvana, Thai G...","Chinese, North Indian, Thai",800,"[('Rated 4.0', 'RATED\n Had been here for din...",[],Buffet,Banashankari
2,https://www.zomato.com/SanchurroBangalore?cont...,"1112, Next to KIMS Medical College, 17th Cross...",San Churro Cafe,Yes,No,3.8/5,918,+91 9663487993,Banashankari,"Cafe, Casual Dining","Churros, Cannelloni, Minestrone Soup, Hot Choc...","Cafe, Mexican, Italian",800,"[('Rated 3.0', ""RATED\n Ambience is not that ...",[],Buffet,Banashankari
3,https://www.zomato.com/bangalore/addhuri-udupi...,"1st Floor, Annakuteera, 3rd Stage, Banashankar...",Addhuri Udupi Bhojana,No,No,3.7/5,88,+91 9620009302,Banashankari,Quick Bites,Masala Dosa,"South Indian, North Indian",300,"[('Rated 4.0', ""RATED\n Great food and proper...",[],Buffet,Banashankari
4,https://www.zomato.com/bangalore/grand-village...,"10, 3rd Floor, Lakshmi Associates, Gandhi Baza...",Grand Village,No,No,3.8/5,166,+91 8026612447\r\n+91 9901210005,Basavanagudi,Casual Dining,"Panipuri, Gol Gappe","North Indian, Rajasthani",600,"[('Rated 4.0', 'RATED\n Very good restaurant ...",[],Buffet,Banashankari


In [3]:
# Check dataset shape (rows, columns)
print("Dataset shape:", df.shape)

Dataset shape: (51717, 17)


## 4. Initial Data Inspection

In [4]:
# Display basic information about the dataset
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 51717 entries, 0 to 51716
Data columns (total 17 columns):
 #   Column                       Non-Null Count  Dtype
---  ------                       --------------  -----
 0   url                          51717 non-null  str  
 1   address                      51717 non-null  str  
 2   name                         51717 non-null  str  
 3   online_order                 51717 non-null  str  
 4   book_table                   51717 non-null  str  
 5   rate                         43942 non-null  str  
 6   votes                        51717 non-null  int64
 7   phone                        50509 non-null  str  
 8   location                     51696 non-null  str  
 9   rest_type                    51490 non-null  str  
 10  dish_liked                   23639 non-null  str  
 11  cuisines                     51672 non-null  str  
 12  approx_cost(for two people)  51371 non-null  str  
 13  reviews_list                 51717 non-null  str  
 14  m

In [5]:
# Check for missing values
df.isnull().sum()

url                                0
address                            0
name                               0
online_order                       0
book_table                         0
rate                            7775
votes                              0
phone                           1208
location                          21
rest_type                        227
dish_liked                     28078
cuisines                          45
approx_cost(for two people)      346
reviews_list                       0
menu_item                          0
listed_in(type)                    0
listed_in(city)                    0
dtype: int64

In [6]:
# Summary statistics for numerical columns
df.describe()

,votes
count,51717.000000
mean,283.697527
std,803.838853
min,0.000000
25%,7.000000
50%,41.000000
75%,198.000000
max,16832.000000


In [7]:
# Preview column names
df.columns

Index(['url', 'address', 'name', 'online_order', 'book_table', 'rate', 'votes',
       'phone', 'location', 'rest_type', 'dish_liked', 'cuisines',
       'approx_cost(for two people)', 'reviews_list', 'menu_item',
       'listed_in(type)', 'listed_in(city)'],
      dtype='str')

In [8]:
# Check for duplicate rows
df.duplicated().sum()

np.int64(0)

### Observations

- The dataset contains 51,717 entries and 17 columns.

- Most columns are categorical (object type), while only "votes" is numerical at this stage.

- The "rate" column is currently stored as a string (e.g., "4.1/5"), which will require cleaning before analysis.

- Missing values are present in several key columns:
  - "rate" has 7,775 missing values
  - "dish_liked" has a significant number of missing values (28,078)
  - "rest_type" and "cuisines" also contain some missing data

- No duplicate rows were found in the dataset.

- The "votes" column shows a wide distribution, with a maximum value of 16,832, indicating the presence of highly popular restaurants.

### Key Data Challenges

- The "rate" column needs to be converted to a numerical format.
- High missing values in "dish_liked" may limit its usefulness.
- Some categorical columns may require standardization or grouping.

## 5. Data Cleaning 

In [9]:
# Function to simplify cuisine categories
def categorize_cuisine(cuisine):
    if pd.isna(cuisine):
        return "Unknown"

    cuisine = cuisine.lower()

    if "indian" in cuisine:
        return "Indian"
    elif "chinese" in cuisine:
        return "Chinese"
    elif "italian" in cuisine:
        return "Italian"
    elif "cafe" in cuisine or "coffee" in cuisine:
        return "Cafe"
    elif "fast food" in cuisine or "burger" in cuisine:
        return "Fast Food"
    else:
        return "Other"

In [10]:
# Apply categorization
df["main_cuisine"] = df["cuisines"].apply(categorize_cuisine)

# Verify new categories
df["main_cuisine"].value_counts()

main_cuisine
Indian       25161
Other        12419
Fast Food     4762
Chinese       3544
Cafe          3443
Italian       2343
Unknown         45
Name: count, dtype: int64

In [11]:
# Clean the 'rate' column by removing '/5' and handling invalid values
def clean_rate(rate):
    if pd.isna(rate):
        return None

    rate = str(rate).strip()

    if rate in ["NEW", "-", ""]:
        return None

    return float(rate.split('/')[0])

In [12]:
# Apply cleaning
df["rate"] = df["rate"].apply(clean_rate)

In [13]:
# Validate result
print("Data type:", df["rate"].dtype)
print("Missing values:", df["rate"].isnull().sum())

Data type: float64
Missing values: 10052


In [14]:
# Clean 'approx_cost(for two people)' column
def clean_cost(cost):
    if pd.isna(cost):
        return None

    cost = str(cost).replace(",", "").strip()

    if cost == "":
        return None

    return float(cost)

In [15]:
# Apply cleaning
df["approx_cost(for two people)"] = df["approx_cost(for two people)"].apply(clean_cost)

In [16]:
# Preview
df["approx_cost(for two people)"].head()

0    800.0
1    800.0
2    800.0
3    300.0
4    600.0
Name: approx_cost(for two people), dtype: float64

### Cleaning Summary

The dataset has been standardized to make it suitable for analysis.

The following transformations were applied:
- cuisine values were grouped into broader categories
- rating values were converted from strings to numeric format
- cost values were cleaned and converted to numeric format

The next step is to narrow the dataset to the most relevant features and restaurants for the business objective.

## 6. Filter Relevant Data

### Selected Relevant Features

Based on the project objectives, the following columns were selected for analysis:

- **name** → restaurant identification
- **location** → neighborhood-level analysis
- **cuisines** → identify Indian restaurants
- **rate** → customer satisfaction
- **votes** → popularity indicator
- **approx_cost(for two people)** → pricing benchmark
- **rest_type** → restaurant category (casual dining, quick bites, etc.)

In [17]:
# Select relevant columns for analysis
selected_columns = [
    "name",
    "location",
    "cuisines",
    "main_cuisine",
    "rate",
    "votes",
    "approx_cost(for two people)",
    "rest_type"
]

df_selected = df[selected_columns].copy()

# Preview selected dataset
df_selected

,name,location,cuisines,main_cuisine,rate,votes,approx_cost(for two people),rest_type
0,Jalsa,Banashankari,"North Indian, Mughlai, Chinese",Indian,4.1,775,800.0,Casual Dining
1,Spice Elephant,Banashankari,"Chinese, North Indian, Thai",Indian,4.1,787,800.0,Casual Dining
2,San Churro Cafe,Banashankari,"Cafe, Mexican, Italian",Italian,3.8,918,800.0,"Cafe, Casual Dining"
3,Addhuri Udupi Bhojana,Banashankari,"South Indian, North Indian",Indian,3.7,88,300.0,Quick Bites
4,Grand Village,Basavanagudi,"North Indian, Rajasthani",Indian,3.8,166,600.0,Casual Dining
...,...,...,...,...,...,...,...,...
51712,Best Brews - Four Points by Sheraton Bengaluru...,Whitefield,Continental,Other,3.6,27,1500.0,Bar
51713,Vinod Bar And Restaurant,Whitefield,Finger Food,Other,NaN,0,600.0,Bar
51714,Plunge - Sheraton Grand Bengaluru Whitefield H...,Whitefield,Finger Food,Other,NaN,0,2000.0,Bar
51715,Chime - Sheraton Grand Bengaluru Whitefield Ho...,"ITPL Main Road, Whitefield",Finger Food,Other,4.3,236,2500.0,Bar


The dataset now contains 51,717 rows and 8 relevant columns.

This confirms that no rows were removed at this stage, only irrelevant features were excluded to focus the analysis on key business variables.

### Secondary Columns

- **dish_liked** → qualitative insights (limited due to missing values)
- **reviews_list** → potential for text analysis (if time allows)

### Columns excluded

- **url, address, phone** → not relevant for analysis
- **menu_item** → too sparse / not useful for current scope

In [19]:
# Filter only Indian restaurants
df_indian = df_selected[df_selected["main_cuisine"] == "Indian"].copy()

# Preview result
print("Indian restaurants dataset shape:", df_indian.shape)
df_indian.head()

Indian restaurants dataset shape: (25161, 8)


,name,location,cuisines,main_cuisine,rate,votes,approx_cost(for two people),rest_type
0,Jalsa,Banashankari,"North Indian, Mughlai, Chinese",Indian,4.1,775,800.0,Casual Dining
1,Spice Elephant,Banashankari,"Chinese, North Indian, Thai",Indian,4.1,787,800.0,Casual Dining
3,Addhuri Udupi Bhojana,Banashankari,"South Indian, North Indian",Indian,3.7,88,300.0,Quick Bites
4,Grand Village,Basavanagudi,"North Indian, Rajasthani",Indian,3.8,166,600.0,Casual Dining
5,Timepass Dinner,Basavanagudi,North Indian,Indian,3.8,286,600.0,Casual Dining


In [20]:
# Keep only the most relevant columns for analysis
columns_to_keep = [
    "name",
    "location",
    "main_cuisine",
    "rate",
    "votes",
    "approx_cost(for two people)",
    "rest_type"
]
# Create analysis dataset with only relevant columns (excluding non-useful fields like url, address, phone)
df_indian = df_indian[columns_to_keep].copy()

# Final preview
df_indian.head()

,name,location,main_cuisine,rate,votes,approx_cost(for two people),rest_type
0,Jalsa,Banashankari,Indian,4.1,775,800.0,Casual Dining
1,Spice Elephant,Banashankari,Indian,4.1,787,800.0,Casual Dining
3,Addhuri Udupi Bhojana,Banashankari,Indian,3.7,88,300.0,Quick Bites
4,Grand Village,Basavanagudi,Indian,3.8,166,600.0,Casual Dining
5,Timepass Dinner,Basavanagudi,Indian,3.8,286,600.0,Casual Dining


In [21]:
# Check missing values in the filtered dataset
df_indian.isnull().sum()

name                              0
location                          0
main_cuisine                      0
rate                           4901
votes                             0
approx_cost(for two people)     144
rest_type                       124
dtype: int64

In [22]:
# Remove rows with missing values in key analytical variables
df_indian = df_indian[
    df_indian["rate"].notna() &
    df_indian["approx_cost(for two people)"].notna()
].copy()

# Check final shape
print("Cleaned Indian dataset shape:", df_indian.shape)

Cleaned Indian dataset shape: (20150, 7)


## 7. Exploratory Data Analysis

In [25]:
df_indian.head(10) # Quick overview of the cleaned dataset

,name,location,main_cuisine,rate,votes,approx_cost(for two people),rest_type
0,Jalsa,Banashankari,Indian,4.1,775,800.0,Casual Dining
1,Spice Elephant,Banashankari,Indian,4.1,787,800.0,Casual Dining
3,Addhuri Udupi Bhojana,Banashankari,Indian,3.7,88,300.0,Quick Bites
4,Grand Village,Basavanagudi,Indian,3.8,166,600.0,Casual Dining
5,Timepass Dinner,Basavanagudi,Indian,3.8,286,600.0,Casual Dining
6,Rosewood International Hotel - Bar & Restaurant,Mysore Road,Indian,3.6,8,800.0,Casual Dining
20,The Vintage Cafe,Banashankari,Indian,3.0,62,400.0,Cafe
26,Coffee Tindi,Banashankari,Indian,3.8,75,200.0,"Cafe, Quick Bites"
29,Srinathji's Cafe,Basavanagudi,Indian,3.8,148,550.0,"Casual Dining, Cafe"
31,Foodiction,Banashankari,Indian,2.8,506,500.0,Quick Bites


In [26]:
df_indian.describe()

,rate,votes,approx_cost(for two people)
count,20150.000000,20150.000000,20150.00000
mean,3.640303,360.173449,628.57995
std,0.442341,917.027275,466.61296
min,1.800000,0.000000,40.00000
25%,3.300000,21.000000,300.00000
50%,3.700000,75.000000,500.00000
75%,3.900000,278.750000,800.00000
max,4.900000,16832.000000,5000.00000


In [27]:
df_indian.info()

<class 'pandas.DataFrame'>
Index: 20150 entries, 0 to 51716
Data columns (total 7 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   name                         20150 non-null  str    
 1   location                     20150 non-null  str    
 2   main_cuisine                 20150 non-null  str    
 3   rate                         20150 non-null  float64
 4   votes                        20150 non-null  int64  
 5   approx_cost(for two people)  20150 non-null  float64
 6   rest_type                    20052 non-null  str    
dtypes: float64(2), int64(1), str(4)
memory usage: 1.2 MB


In [28]:
df_indian["rate"].describe()

count    20150.000000
mean         3.640303
std          0.442341
min          1.800000
25%          3.300000
50%          3.700000
75%          3.900000
max          4.900000
Name: rate, dtype: float64

In [ ]:
#in process...

## 8. Insights